rkovalenko/ReverseBTree.cs

```diff
@@ -0,0 +1,33 @@
+/**
+ * Definition for a binary tree node.
+ * public class TreeNode {
+ *     public int val;
+ *     public TreeNode left;
+ *     public TreeNode right;
+ *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
+ *         this.val = val;
+ *         this.left = left;
+ *         this.right = right;
+ *     }
+ * }
+ */
+public class Solution {
+    public TreeNode InvertTree(TreeNode root) {
+        return RevertNode(root);
+    }
+    
+    public TreeNode RevertNode(TreeNode node)
+    {
+        if(node is not null)
+        {
+            var temp = node.left;
+            node.left = node.right;
+            node.right = temp;
+            
+            RevertNode(node.left);
```

Comment:
Okay for some reason recursion works better in those types of tasks. :D I'm the slowest one so far. Nice.

Also here I think it may appear less optimal because of using PreOrderTraversal instead of PostOrder.